src/main/java/org/junitpioneer/jupiter/issue/IssueExtensionListener.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.issue;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.junit.platform.engine.TestExecutionResult;
+import org.junit.platform.engine.reporting.ReportEntry;
+import org.junit.platform.launcher.TestExecutionListener;
+import org.junit.platform.launcher.TestIdentifier;
+import org.junit.platform.launcher.TestPlan;
+
+/**
+ * <p>This listener collects the names and results of all tests, which are annotated with the {@link org.junitpioneer.jupiter.Issue} annotation.
+ * After all tests are finished the results are provided to an {@link IssueProcessor} for further processing.</p>
+ */
+public class IssueExtensionListener implements TestExecutionListener {
+
+	static final String KEY_ISSUE = "Issue";
+
+	// Storage with all tests that belong to an issue <issueId, List<UniqueIdentifier>>
+	private final ConcurrentHashMap<String, List<String>> issueTestsStorage = new ConcurrentHashMap<>();
+
+	// Storage with tests results of test cases <UniqueIdentifier, result>
+	private final ConcurrentHashMap<String, String> testStatusStorage = new ConcurrentHashMap<>();
```

Comment:
This extension isn't stateless. As far as I'm aware, Jupiter doesn't guarantee it will use the same instance for the entire test run (although I couldn't verify that in the user guide). I wanted to propose to use the store to store and retrieve all necessary information, but I just see that `TestExecutionListener`s don't have access to that (makes sense, because `TestExecutionListener` doesn't know anything about Jupiter, which is where the store lives).